wait-for.c

```diff
@@ -245,21 +243,21 @@ int main(int argc, const char **argv) {
 		goto exit;
 	}
 
-	if (args.username == NULL) {
-		// attempt to populate based on current UID
-		struct passwd *pwd = getpwuid(getuid());
-		if (pwd == NULL) {
+	errno = 0;
```

Comment:
This is generally considered bad practice. What are you trying to do?

Reply:
Nope, it's necessary:

Quote man page errno(3):

> In such cases, a successful return can  be distinguished from an error return by setting errno to zero before the call [...]

Quote man page getpwnam(3):

>If one wants to check errno after the call, it should be set to zero before the call

Reply:
I'll cede it's necessary in the case of `getpwnam` but it's still generally considered bad practice. I understand it's entirely possible, but it usually means you're hiding an error somewhere and thus it's discouraged.